src/core/resources/_selectors/assets.ts

```diff
@@ -104,10 +104,15 @@ export function selectUserAssetWithUniqueId(uniqueId: UniqueId) {
   };
 }
 
-export function selectUserAssetsBalance(assets: ParsedAssetsDictByChain) {
+export function selectUserAssetsBalance(
+  assets: ParsedAssetsDictByChain,
+  hidden: (asset: ParsedUserAsset) => boolean,
+) {
   const networksTotalBalance = Object.values(assets).map((assetsOnject) => {
     const assetsNetwork = Object.values(assetsOnject);
+
     const networkBalance = assetsNetwork
+      .filter((asset) => !hidden(asset))
       .map((asset) => asset.native.balance.amount)
```

Comment:
You technically could use `hiddenStore.getState()` and then filter based off of the array, but that won't cause a re-render if we were to remove tokens from another component while this one stays as is.